impl/spanner/mutations/table_test.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mutations
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/keytransparency/core/adminserver"
+	"github.com/google/keytransparency/core/integration/storagetest"
+	"github.com/google/keytransparency/core/keyserver"
+	"github.com/google/keytransparency/core/water"
+	"github.com/google/keytransparency/impl/spanner/directory"
+	"github.com/google/keytransparency/impl/spanner/testutil"
+	"github.com/google/trillian/crypto/keyspb"
+	"google.golang.org/protobuf/proto"
+
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
+	dtype "github.com/google/keytransparency/core/directory"
+	ktspanner "github.com/google/keytransparency/impl/spanner"
+	tpb "github.com/google/trillian"
+)
+
+func mustMarshal(t *testing.T, p proto.Message) []byte {
+	t.Helper()
+	b, err := proto.Marshal(p)
+	if err != nil {
+		t.Errorf("proto.Marshal(): %v", err)
+	}
+	return b
+}
+
+func NewForTest(ctx context.Context, t testing.TB, dirID string, logIDs ...int64) *Table {
+	t.Helper()
+	ddl, err := ktspanner.ReadDDL()
+	if err != nil {
+		t.Fatal(err)
+	}
+	client := testutil.CreateDatabase(ctx, t, ddl)
+	q := New(client)
+
+	if err := directory.New(client).Write(ctx, &dtype.Directory{
+		DirectoryID: dirID,
+		Map:         &tpb.Tree{},
+		Log:         &tpb.Tree{},
+		VRFPriv:     &keyspb.PrivateKey{Der: []byte("privkeybytes")},
+	}); err != nil {
+		t.Fatalf("directories.Write(%v): %v", dirID, err)
+	}
+	if err := q.AddLogs(ctx, dirID, logIDs...); err != nil {
+		t.Fatalf("AddLogs(): %v", err)
+	}
+	time.Sleep(readStaleness)
+
+	return q
+}
+
+func TestNewForTest(t *testing.T) {
+	ctx := context.Background()
+	NewForTest(ctx, t, "NewForTest", 1)
```

Comment:
Supernit: the other directory IDs are all lowercase. Can they be consistent in one direction or the other?